hazelcast/src/test/java/com/hazelcast/client/topic/ClientTopicTest.java

```diff
@@ -93,4 +107,117 @@ public void testGetLocalTopicStats() throws Exception {
 
         topic.getLocalTopicStats();
     }
+
+    @Test
+    public void testPublish() throws InterruptedException {
+        String publishValue = "message";
+        ITopic<String> topic = client.getTopic(randomString());
+        final AtomicInteger count = new AtomicInteger(0);
+        final Collection<String> receivedValues = new ArrayList<>();
+
+        topic.addMessageListener(new MessageListener<String>() {
+
+            @Override
+            public void onMessage(Message<String> message) {
+                count.incrementAndGet();
+                receivedValues.add(message.getMessageObject());
+            }
+        });
+        topic.publish(publishValue);
+        assertTrueEventually(new AssertTask() {
+            @Override
+            public void run() {
+                assertEquals(1, count.get());
+                assertTrue(receivedValues.contains(publishValue));
+            }
+        });
+    }
+
+
+    @Test
+    public void testPublishAsync() throws InterruptedException {
+        ITopic<String> topic = client.getTopic(randomString());
+        final AtomicInteger count = new AtomicInteger(0);
+        final List<String> receivedValues = new ArrayList<>();
+
+        topic.addMessageListener(new MessageListener<String>() {
+
+            @Override
+            public void onMessage(Message<String> message) {
+                count.incrementAndGet();
+                receivedValues.add(message.getMessageObject());
+            }
+        });
+        final String message = "message";
+        topic.publishAsync(message);
+        assertTrueEventually(new AssertTask() {
+            @Override
+            public void run() {
+                assertEquals(1, count.get());
+                assertEquals(Arrays.asList(message), receivedValues);
+            }
+        });
+    }
+
+    @Test
+    public void testPublishAll() throws InterruptedException {
```

Comment:
`testPublishAll` and `testPublishAllAsync` are the same. Can you fix that? I guess they can share a single helper method to avoid duplication.

Same for other two tests when it comes to avoiding duplication.

Reply:
Oh no, this was a bad copy paste error. -_-